src/decompress.rs

```diff
@@ -296,6 +296,7 @@ impl<'s, 'd> Decompress<'s, 'd> {
                 // to [0, 0]. But the last copy wrote to [9, 24], which is 24
                 // extra bytes in dst *beyond* the end of the copy, which is
                 // guaranteed by the conditional above.
+                let dest_len = self.dst.len();
```

Comment:
I might just add a very short comment here saying something like, "Save destination length here to avoid a reborrow UB violation under the Tree Borrows model."